Exiled.Events/Patches/Events/Player/FirearmRequestReceived.cs

```diff
@@ -263,12 +265,40 @@ private static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructi
                     new(OpCodes.Stloc_S, 6),
                 });
 
-            newInstructions[newInstructions.Count - 1].WithLabels(returnLabel);
+            offset = -3;
+            index = newInstructions.FindLastIndex(instruction => instruction.opcode == OpCodes.Ldftn) + offset;
+
+            newInstructions.InsertRange(
+                index,
+                new[]
+                {
+                    // player
+                    new CodeInstruction(OpCodes.Ldloc_S, player.LocalIndex),
+
+                    // firearm
+                    new(OpCodes.Ldloc_S, firearm.LocalIndex),
+
+                    // InspectingWeaponEventArgs ev = new(Player, firearm)
```

Comment:
You haven't tested your PR after performing the requested changes, as this will throw an exception due to missing `bool` argument to the `InspectingWeaponEventArgs::cctor`

Reply:
holy shit i was accidentally testing my old version sorry, one second

Reply:
i cant test this version bc im on mobile but lmk if the build fails or some shit